tools/openms/BaselineFilter.xml

```diff
@@ -1,76 +1,83 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <!--This is a configuration file for the integration of a tools into Galaxy (https://galaxyproject.org/). This file was automatically generated using CTDConverter.-->
 <!--Proposed Tool Section: [Signal processing and preprocessing]-->
-<tool id="BaselineFilter" name="BaselineFilter" version="2.3.0">
+<tool id="BaselineFilter" name="BaselineFilter" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@">
   <description>Removes the baseline from profile spectra using a top-hat filter.</description>
   <macros>
     <token name="@EXECUTABLE@">BaselineFilter</token>
     <import>macros.xml</import>
+    <import>macros_autotest.xml</import>
+    <import>macros_test.xml</import>
   </macros>
-  <expand macro="references"/>
-  <expand macro="stdio"/>
   <expand macro="requirements"/>
-  <command detect_errors="aggressive"><![CDATA[BaselineFilter
-
-#if $param_in:
-  -in $param_in
-#end if
-#if $param_out:
-  -out $param_out
-#end if
-#if $param_struc_elem_length:
-  -struc_elem_length $param_struc_elem_length
+  <expand macro="stdio"/>
+  <command detect_errors="exit_code"><![CDATA[@QUOTE_FOO@
+@EXT_FOO@
+#import re
+mkdir param_in &&
+ln -s '$param_in' 'param_in/${re.sub("[^\w\-_]", "_", $param_in.element_identifier)}.$gxy2omsext($param_in.ext)' &&
+mkdir param_out_out &&
+@EXECUTABLE@
+-in
+'param_in/${re.sub("[^\w\-_]", "_", $param_in.element_identifier)}.$gxy2omsext($param_in.ext)'
```

Comment:
Oh, you are still constructing the cmd? Did the configfile approach not worked out? For example, those replacements could be avoided in the config file mode I believe. 

I will try to look at this in more detail on Wednesday!

Thanks @bernt-matthias!

Reply:
> Did the configfile approach not worked out? For example, those replacements could be avoided in the config file mode I believe.

Unfortunately the replacements seem inevitable:

- It seems that input files can not be specified in the (ini) config files at all. I just tried it. Other indicators that this is actually the case are 1) the `type="input-file"` tags don't have a value field 2) The tests described in  https://github.com/OpenMS/OpenMS/blob/develop/src/tests/topp/CMakeLists.txt that use ini files also don't set the input files in the ini files but on the command line. To be sure we could still ask the developers.
- The extensions is important for determining the data type of input files in many cases. OpenMS implements sniffers for many but not all data types. For those without sniffers the file extension is used (for those with sniffers and all output files `.tmp` would work as extension). Unfortunately these are the "under specified" data types which will require a thorough review also in https://github.com/galaxyproject/galaxy/pull/8109.
- Also for the proper detection of the Galaxy data types of the outputs the extension is used in many cases.

Reply:
With https://github.com/WorkflowConversion/CTDConverter/pull/54/commits/bdf37bf4ab3827c592e44171a8e4b46482f3075c we are on a good way to CTD based input :) 

Still some bits to fix (in particular the test generation).

Reply:
Just awesome!!!